In [13]:
import sys
sys.path.append('..')

from prompt import PROMPT
from PIL import Image

prompt = PROMPT
image = Image.open("../base_image.png")

## nano banana

In [6]:
from google import genai
from google.genai import types

client = genai.Client()

response = client.models.generate_content(
    model="gemini-2.5-flash-image",
    contents=[prompt, image],
)

In [9]:
for part in response.parts:
    if part.text is not None:
        print(part.text)
    elif part.inline_data is not None:
        image = part.as_image()
        image.save("../images/nanoBanana_image.png")

네, 요청하신 내용에 맞춰 이미지를 생성해 드릴게요. 




In [12]:
response.usage_metadata

GenerateContentResponseUsageMetadata(
  candidates_token_count=1306,
  candidates_tokens_details=[
    ModalityTokenCount(
      modality=<MediaModality.IMAGE: 'IMAGE'>,
      token_count=1290
    ),
  ],
  prompt_token_count=514,
  prompt_tokens_details=[
    ModalityTokenCount(
      modality=<MediaModality.TEXT: 'TEXT'>,
      token_count=256
    ),
    ModalityTokenCount(
      modality=<MediaModality.IMAGE: 'IMAGE'>,
      token_count=258
    ),
  ],
  total_token_count=1820
)

Input 비용:  514 / 1,000,000 × $0.30  = $0.000154
Output 비용: 1,306 / 1,000,000 × $30.00 = $0.039180
─────────────────────────────────────────────────────
총 비용:                                ≈ $0.0393

응답 시간: 14.5초

## nano banana pro

In [26]:
from PIL import Image
img = Image.open('../base_image.png')
img = img.resize((1024, 1536))  # 크기 축소

In [29]:
from google import genai
from google.genai import types

aspect_ratio = "9:16" # "1:1","2:3","3:2","3:4","4:3","4:5","5:4","9:16","16:9","21:9"
resolution = "1K" # "1K", "2K", "4K"

client = genai.Client()

response_pro = client.models.generate_content(
    model="gemini-3-pro-image-preview",
    contents=[
        prompt,
        img
    ],
    config=types.GenerateContentConfig(
        response_modalities=['TEXT', 'IMAGE'],
        image_config=types.ImageConfig(
            aspect_ratio=aspect_ratio,
            #image_size=resolution
        ),
    )
)


In [31]:
for part in response_pro.parts:
    if part.text is not None:
        print(part.text)
    elif part.inline_data is not None:
        image = part.as_image()
        image.save("../images/nanoBanana_pro_image.png")

In [32]:
response.usage_metadata

GenerateContentResponseUsageMetadata(
  candidates_token_count=1306,
  candidates_tokens_details=[
    ModalityTokenCount(
      modality=<MediaModality.IMAGE: 'IMAGE'>,
      token_count=1290
    ),
  ],
  prompt_token_count=514,
  prompt_tokens_details=[
    ModalityTokenCount(
      modality=<MediaModality.TEXT: 'TEXT'>,
      token_count=256
    ),
    ModalityTokenCount(
      modality=<MediaModality.IMAGE: 'IMAGE'>,
      token_count=258
    ),
  ],
  total_token_count=1820
)

Input 비용:       514 / 1,000,000 × $2.00   = $0.00103
Output 텍스트:     16 / 1,000,000 × $12.00  = $0.00019
Output 이미지:  1,290 / 1,000,000 × $120.00 = $0.15480
─────────────────────────────────────────────────────────
총 비용:                                   ≈ $0.156

응답 시간: 20.4초